### PR TITLE
fix: Use NormalizedTextConfigWithGQA for Qwen2, Qwen3, and Qwen3 MoE

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Fixes ONNX export for Qwen2, Qwen3, and Qwen3 MoE models by using `NormalizedTextConfigWithGQA` instead of the default `NormalizedTextConfig` inherited from `LlamaOnnxConfig`.

## Problem

Qwen2, Qwen3, and Qwen3 MoE models use GQA where `num_key_value_heads` != `num_attention_heads`. When exporting to ONNX, users encounters dimension mismatch errors: 

```
INVALID_ARGUMENT: Got invalid dimensions for input: past_key_values.9.value
index: 3 Got: 64 Expected: 128
```

## Root Cause

These config classes inherit from `LlamaOnnxConfig`, which uses `NORMALIZED_CONFIG_CLASS = NormalizedTextConfig`. This doesn't properly handle GQA architectures, leading to incorrect head dim calculations in the KV cache.

**Example (Qwen2.5-7B-Instruct):**

- `num_attention_heads: 28`
- `num_key_value_heads: 4` (GQA with 7:1 ratio)
- `hidden_size: 3584`
- `head_dim: 128` (calculated)

## Solution

Override `NORMALIZED_CONFIG_CLASS = NormalizedTextConfigWithGQA` in:
- `Qwen2OnnxConfig`
- `Qwen3OnnxConfig`
- `Qwen3MoeOnnxConfig`

This follows the same pattern as `ArceeOnnxConfig` which also inherits from `LlamaOnnxConfig` but correctly override the normalized config class.

## Changes

**File: `optimum/exporters/onnx/model_configs.py`**

Added `NORMALIZED_CONFIG_CLASS = NormalizedTextConfigWithGQA` to three config classes:

```python
@register_tasks_manager_onnx("qwen2", ...)
class Qwen2OnnxConfig(LlamaOnnxConfig):
    NORMALIZED_CONFIG_CLASS = NormalizedTextConfigWithGQA  # Added
    MIN_TRANSFORMERS_VERSION = version.parse("4.37.0")

@register_tasks_manager_onnx("qwen3", ...)
class Qwen3OnnxConfig(LlamaOnnxConfig):
    NORMALIZED_CONFIG_CLASS = NormalizedTextConfigWithGQA  # Added
    MIN_TRANSFORMERS_VERSION = version.parse("4.51.0")

@register_tasks_manager_onnx("qwen3_moe", ...)
class Qwen3MoeOnnxConfig(LlamaOnnxConfig):
    NORMALIZED_CONFIG_CLASS = NormalizedTextConfigWithGQA  # Added
    MIN_TRANSFORMERS_VERSION = version.parse("4.51.0")
    _MODEL_PATCHER = Qwen3MoeModelPatcher
```

### Test 1: config classes
Verified all three config classes now use `NormalizedTextConfigWithGQA`:
```
Qwen2OnnxConfig uses NormalizedTextConfigWithGQA
Qwen3OnnxConfig uses NormalizedTextConfigWithGQA
Qwen3MoeOnnxConfig uses NormalizedTextConfigWithGQA
```

### Test 2: Architecture Verification 
**Qwen2.5-7B-Instruct:**
```
model_type: qwen2
num_attention_heads: 28
num_key_value_heads: 4
hidden_size: 3584
head_dim: 128 (calculated)
Uses GQA (num_kv_heads=4 != num_attn_heads=28)
Correct head_dim = 128
```

**Qwen3-235B-A22B (MoE):**
```
model_type: qwen3_moe
num_attention_heads: 64
num_key_value_heads: 4
hidden_size: 4096
head_dim: 128
Uses GQA (16:1 ratio)
```

### Test 3: ONNX Config
```
Normalized class: NormalizedTextConfigWithGQA
Using correct normalized config
```

### Summary

- [x] All three config classes use correct normalized config
- [x] Verified GQA architecture in Qwen2.5-7B and Qwen3-235B-A22B MoE
- [x] ONNX config properly uses `NormalizedTextConfigWithGQA`
- [x] Head dimensions calculated correctly (128)
- [x] Follows same pattern as `ArceeOnnxConfig`

## Related

Fixes huggingface/optimum#2379

@IlyasMoutawwakil

### CODE REVIEW

**Comment by Aznix07** [comment]
Hi @IlyasMoutawwakil,

I ran the Qwen3 tests as requested. **All tests passed** 

## Environment

- **Python:** 3.13.7
- **Platform:** Windows 11
- **pytest:** 9.0.0
- **Virtual environment:** Python venv

## Commands Run

```bash
python -m venv .venv
.venv\Scripts\activate
pip install -e .[tests,onnxruntime]
python -m pytest tests/onnxruntime/test_decoder.py -k "qwen3" -v
```

## Test Results

```
========================= test session starts ==========================
platform win32 -- Python 3.13.7, pytest-9.0.0, pluggy-1.6.0
cachedir: .pytest_cache
rootdir: optimum-onnx
configfile: pyproject.toml
plugins: anyio-4.11.0, xdist-3.8.0
collected 372 items / 356 deselected / 16 selected

tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_beam_search_to_transformers_36_qwen3_True PASSED [  6%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_beam_search_to_transformers_37_qwen3_moe_True PASSED [ 12%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_generation_to_transformers_36_qwen3_True PASSED [ 18%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_generation_to_transformers_37_qwen3_moe_True PASSED [ 25%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_generation_with_and_without_io_binding_36_qwen3_True PASSED [ 31%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_generation_with_and_without_io_binding_37_qwen3_moe_True PASSED [ 37%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_generation_with_and_without_past_key_values_36_qwen3 PASSED [ 43%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_generation_with_and_without_past_key_values_37_qwen3_moe PASSED [ 50%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_to_transformers_72_qwen3_True PASSED [ 56%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_to_transformers_73_qwen3_False PASSED [ 62%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_to_transformers_74_qwen3_moe_True PASSED [ 68%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_to_transformers_75_qwen3_moe_False PASSED [ 75%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_with_and_without_io_binding_72_qwen3_True PASSED [ 81%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_with_and_without_io_binding_73_qwen3_False PASSED [ 87%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_with_and_without_io_binding_74_qwen3_moe_True PASSED [ 93%]
tests/onnxruntime/test_decoder.py::ORTModelForCausalLMIntegrationTest::test_compare_logits_with_and_without_io_binding_75_qwen3_moe_False PASSED [100%]

===== 16 passed, 356 deselected, 32 warnings in 214.37s (0:03:34) ======
```

## Summary

✅ All 16 Qwen3/Qwen3-MoE tests passed successfully

Test Covered:

- ✅ Beam search generation (Qwen3 & Qwen3-MoE)
- ✅ Standard generation (Qwen3 & Qwen3-MoE)
- ✅ Generation with/without IO binding (Qwen3 & Qwen3-MoE)
- ✅ Generation with/without past key values (Qwen3 & Qwen3-MoE)
- ✅ Logits comparison to transformers (Qwen3 & Qwen3-MoE)
- ✅ Logits with/without IO binding (Qwen3 & Qwen3-MoE)

The fix correctly handles GQA (Grouped Query Attention) for both Qwen3 and Qwen3 MoE models by using NormalizedTextConfigWithGQA instead of NormalizedTextConfig.

Let me know if you need any other tests or changes!

Thanks!

**Comment by IlyasMoutawwakil** [comment]
Thanks for the fix ! merging !
